networkit/cpp/distance/test/FloydWarshallGTest.cpp

```diff
@@ -0,0 +1,415 @@
+                << "source " << source << " target " << target;
+        }
+    }
+}
+
+TEST_F(FloydWarshallGTest, testGetNodesOnShortestPathDisconnectedGraph) {
+    auto graph = disconnectedGraph();
+    FloydWarshall test(graph);
+    test.run();
+    const std::vector<std::vector<std::vector<node>>> expectedNodesOnShortestPaths{
+        {{0}, {0, 1}, {0, 1, 2}, {}},
+        {{1, 0}, {1}, {1, 2}, {}},
+        {{2, 1, 0}, {2, 1}, {2}, {}},
+        {{}, {}, {}, {3}}};
+    for (node source = 0; source < graph.numberOfNodes(); ++source) {
+        for (node target = 0; target < graph.numberOfNodes(); ++target) {
+            EXPECT_EQ(test.getNodesOnShortestPath(source, target),
+                      expectedNodesOnShortestPaths[source][target])
+                << source << ", " << target;
+        }
+    }
+}
+
+TEST_F(FloydWarshallGTest, testIsNodeInNegativeCycleDisconnectedGraph) {
+    auto graph = disconnectedGraph();
+    FloydWarshall test(graph);
+    test.run();
+    for (node source = 0; source < graph.numberOfNodes(); ++source) {
+        EXPECT_FALSE(test.isNodeInNegativeCycle(source));
+    }
+}
+
+TEST_F(FloydWarshallGTest, testGetDistanceDirectedGraphWithNegativeSelfLoop) {
+    auto graph = directedGraphWithNegativeSelfLoop();
+    FloydWarshall test(graph);
+    test.run();
+    constexpr edgeweight expectedDistance{-std::numeric_limits<edgeweight>::infinity()};
+    for (node source = 0; source < graph.numberOfNodes(); ++source) {
+        for (node target = 0; target < graph.numberOfNodes(); ++target) {
+            EXPECT_EQ(test.getDistance(source, target), expectedDistance);
+        }
+    }
+}
+
+TEST_F(FloydWarshallGTest, testIsNodeInNegativeCycleDirectedGraphWithNegativeSelfLoop) {
+    auto graph = directedGraphWithNegativeSelfLoop();
+    FloydWarshall test(graph);
+    test.run();
+    for (node source = 0; source < graph.numberOfNodes(); ++source) {
+        EXPECT_TRUE(test.isNodeInNegativeCycle(source));
+    }
+}
+
+TEST_F(FloydWarshallGTest, testMultipleShortestDistancePaths) {
+    Graph graph(11, true);
+    // Shortest path, first case [0,10] with 5 nodes (inclusive)
+    graph.addEdge(0, 1, 1);
+    graph.addEdge(1, 2, 1);
+    graph.addEdge(2, 3, 1);
+    graph.addEdge(3, 10, 2);
+    // Shortest path, second case [0,10] with 4 nodes (inclusive)
+    graph.addEdge(0, 4, 1);
+    graph.addEdge(4, 5, 1);
+    graph.addEdge(5, 10, 3);
+    // Shortest path, third case [0,10] with 6 nodes (inclusive)
+    graph.addEdge(0, 6, 1);
+    graph.addEdge(6, 7, 1);
+    graph.addEdge(7, 8, 1);
+    graph.addEdge(8, 9, 1);
+    graph.addEdge(9, 10, 1);
+    FloydWarshall test(graph);
+    test.run();
+    constexpr edgeweight expectedShortestDistance = 5.0;
+    const std::vector<node> expectedPath{0, 4, 5, 10};
+    EXPECT_EQ(test.getDistance(0, 10), expectedShortestDistance);
+    EXPECT_EQ(test.getNodesOnShortestPath(0, 10), expectedPath);
+}
+
+TEST_F(FloydWarshallGTest, testGetDistanceWithMediumSizedGraph) {
```

Comment:
As a reader, I find this test and the one below too complicated to follow. Tests should be like the ones you wrote above:
- Define an instance
- Run the algorithm
- Check the result.

These two tests have too much additional logic. In case they start failing, maintainers will have to understand what exactly they are testing before trying to fix the bug. I suggest to either simplify those tests (e.g., by using a smaller instance) or removw them.